### PR TITLE
fix texture memory leak on Metal renderer.

### PR DESCRIPTION
**Pull request by xianyinchen**
Just switch two scene will reproduce this issue.
[NewProject_2.zip](https://github.com/cocos-creator/engine-native/files/6654522/NewProject_2.zip)

### CODE REVIEW

**Comment by zhefengzhang** [comment]
关联 issues：https://github.com/cocos-creator/engine-native/pull/3621#issue-670239064

File: cocos/renderer/gfx-metal/MTLTexture.mm [hunk 1]
```diff
@@ -131,6 +131,7 @@ of this software and associated engine source code (the "Software"), a limited,
 
     std::function<void(void)> destroyFunc = [=]() {
         if (mtlTexure) {
+            [mtlTexure setPurgeableState:MTLPurgeableStateEmpty];
```
Comment (minggo) [inline cocos/renderer/gfx-metal/MTLTexture.mm:134]: Why need to invoke this function?

**Review by hana-alice — APPROVED** [review]
LGTM，maybe same option should be applied to MTLBuffer.

**Comment by minggo** [comment]
This PR includes other commits, please send a new one. Thanks.

**Pull request closed without merging**